crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -175,15 +175,52 @@ impl LogState {
         }
     }
 
+    /// Moves the log state to [`LogState::Available`].
+    ///
+    /// # Panics
+    /// This method panics if you try setting the log state of a newer segment than the provided
+    /// one.
+    fn transition_to_available(
+        &mut self,
+        new_segment_index: SegmentIndex,
+        new_configuration: LogletConfiguration,
+    ) {
+        match self {
+            LogState::Provisioning { .. } => {
+                *self = LogState::Available {
+                    configuration: Some(new_configuration),
+                    segment_index: new_segment_index,
+                }
+            }
+            LogState::Available { segment_index, .. }
+            | LogState::Sealing { segment_index, .. }
+            | LogState::Sealed { segment_index, .. }
+                if *segment_index < new_segment_index =>
+            {
+                *self = LogState::Available {
+                    configuration: Some(new_configuration),
+                    segment_index: new_segment_index,
+                }
+            }
+            _ => {
+                panic!("Trying to transition LogState to Available with a segment that should be sealed.");
```

Comment:
The panic with the conditional check is not quite right because we might have learned from the updated `Logs` configuration about the extended log segment before we receive the `SealAndExtendSucceeded` message.